server/src/main/java/org/opensearch/search/aggregations/bucket/BucketsAggregator.java

```diff
@@ -129,6 +130,14 @@ public final void collectExistingBucket(LeafBucketCollector subCollector, int do
         subCollector.collect(doc, bucketOrd);
     }
 
+    public final void collectStarTreeBucket(StarTreeBucketCollector subCollector, long docCount, long bucketOrd, int entryBit)
+        throws IOException {
+        if (docCounts.increment(bucketOrd, docCount) == docCount) {
+            multiBucketConsumer.accept(0);
+        }
+        subCollector.collectStarEntry(entryBit, bucketOrd);
+    }
```

Comment:
I've been thinking through this and it mostly makes sense. 

I guess my one objection is the use of the word "collect", since it's not really related to collectors (or rather, it doesn't need to be).

Reply:
That said, see my comment above -- I think you could just intercept the whole thing earlier and not have to do a custom collector.